recipes/qt/6.x.x/test_package/conanfile.py

```diff
@@ -0,0 +1,129 @@
+import os
+import shutil
+
+from conans import ConanFile, tools, Meson, RunEnvironment, CMake
+from conans.errors import ConanException
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "qt", "cmake", "cmake_find_package_multi", "cmake_find_package", "pkg_config", "qmake"
+
+    def build_requirements(self):
+        self.build_requires("cmake/3.19.1")
+        if self._meson_supported():
+            self.build_requires("meson/0.56.2")
+
+    def _is_mingw(self):
+        return self.settings.os == "Windows" and self.settings.compiler == "gcc"
+
+    def _meson_supported(self):
+        return False and self.options["qt"].shared and\
+            not tools.cross_building(self.settings) and\
+            not tools.os_info.is_macos and\
+            not self._is_mingw()
+
+    def _qmake_supported(self):
+        return self.settings.compiler != "Visual Studio" or self.options["qt"].shared
+
+    def _cmake_multi_supported(self):
+        return self.settings.compiler != "Visual Studio" or not self.options["qt"].shared or self.settings.build_type != "Debug"
+
+    def _build_with_qmake(self):
+        if not self._qmake_supported():
+            return
+        tools.mkdir("qmake_folder")
+        with tools.chdir("qmake_folder"):
+            self.output.info("Building with qmake")
+
+            with tools.vcvars(self.settings) if self.settings.compiler == "Visual Studio" else tools.no_op():
+                args = [self.source_folder, "DESTDIR=bin"]
+
+                def _getenvpath(var):
+                    val = os.getenv(var)
+                    if val and tools.os_info.is_windows:
+                        val = val.replace("\\", "/")
+                        os.environ[var] = val
+                    return val
+
+                value = _getenvpath('CC')
+                if value:
+                    args += ['QMAKE_CC=' + value,
+                             'QMAKE_LINK_C=' + value,
+                             'QMAKE_LINK_C_SHLIB=' + value]
+
+                value = _getenvpath('CXX')
+                if value:
+                    args += ['QMAKE_CXX=' + value,
+                             'QMAKE_LINK=' + value,
+                             'QMAKE_LINK_SHLIB=' + value]
```

Comment:
Is it not more robust with https://github.com/bincrafters/community/issues/1269#issuecomment-688743713?

Reply:
the messages in this issue do not indicate improvements with https://github.com/ericLemanissier/conan-qt/commit/2d6fd88d049b72c32fc15b358a576b4478a1af0e, so I'm not sure. We might as well remove the qmake test if you think it's brittle

Reply:
oh sure it's an improvement. I have to modify qt recipe with these modifications if CC and CXX are set, otherwise it fails.